types.ts

```diff
@@ -0,0 +1,65 @@
+export interface WeatherData {
+  cnt: number;
+  list: City[];
+}
+
+export interface City {
+  coord: Coord;
+  sys: Sys;
+  weather: Weather[];
+  main: Main;
+  visibility: number;
+  wind: Wind;
+  clouds: Clouds;
+  dt: number;
+  id: number;
+  name: string;
+  rain?: Rain;
+}
+
+interface Coord {
+  lon: number;
+  lat: number;
+}
+
+interface Sys {
+  country: string;
+  sunrise: number;
+  sunset: number;
+}
+
+interface Weather {
+  id: number;
+  main: string;
+  description: string;
+  icon: string;
+}
+
+interface Main {
+  temp: number;
+  feels_like: number;
+  temp_min: number;
+  temp_max: number;
+  pressure: number;
+  humidity: number;
+}
+
+interface Wind {
+  speed: number;
+  deg: number;
+}
+
+interface Clouds {
+  all: number;
+}
+
+interface Rain {
+  "1h"?: number;
```

Comment:
I assume this type is used as a definition of what's coming from API. In this case this property will be returned as number or undefined.
There's a slight difference between optional property and defining something as `undefined`. I think in this case second option is more appropriate. More info here:
https://tkdodo.eu/blog/optional-vs-undefined#optional-properties

```suggestion
  "1h": number | undefined;
```